IN-CORE/incore-services
# 15 handle some inconsistencies in eq demand types and units naming on hazard service

Add checks for demand type and demand units using the table stored in the database. 

To test:
- e.g POST to `http://localhost:8080/hazard/api/earthquakes/5b902cb273c3371e1236b36b/values` with point 
```
[
                {
                    "demands": ["0.2 Sec SA"],
                    "units": ["g"],
                    "loc": "35.84,-89.90"
                }
            ]
```
<img width="882" alt="image" src="https://user-images.githubusercontent.com/13950475/192334567-30fe9efc-5a21-45ce-be51-32e3465c656e.png">


- Varying the demand type and demand unit to see if it complains. 
- Repeat same test with tsunami, hurricane, flood, and tornado
Note:
-  `-9999.2`means invalid demand type
-  `-9999.3` means invalid demand unit
- `-9999.5` is normal because you might not have that hazard tiff file on your local machine


------------
Here are the allowed types and their demands:
```
{
    "_id" : ObjectId("619281abac0262225b2c2f4f"),
    "earthquake" : [
        {
            "demand_type" : "pga",
            "demand_unit" : [
                "g",
                "in/sec^2",
                "m/sec^2"
            ],
            "description" : "Peak ground acceleration"
        },
        {
            "demand_type" : "pgv",
            "demand_unit" : [
                "in/sec",
                "cm/sec"
            ],
            "description" : "Peak ground velocity"
        },
        {
            "demand_type" : "pgd",
            "demand_unit" : [
                "in",
                "ft",
                "m"
            ],
            "description" : "Peak ground displacement"
        },
        {
            "demand_type" : "sa",
            "demand_unit" : [
                "g",
                "in/sec^2",
                "m/sec^2"
            ],
            "description" : "Spectral acceleration"
        },
        {
            "demand_type" : "sd",
            "demand_unit" : [
                "in",
                "ft",
                "m"
            ],
            "description" : "Spectral displacement"
        },
        {
            "demand_type" : "sv",
            "demand_unit" : [
                "cm/s",
                "in/s"
            ],
            "description" : "Spectral Velocity"
        }
    ],
    "tsunami" : [
        {
            "demand_type" : "Hmax",
            "demand_unit" : [
                "ft",
                "m"
            ],
            "description" : "Onshore: maximum tsunami height above local ground level overland. Offshore: maximum tsunami height taken crest to trough"
        },
        {
            "demand_type" : "Vmax",
            "demand_unit" : [
                "mph",
                "kph",
                "ft/sec",
                "m/sec"
            ],
            "description" : "Maximum near-coast or overland water velocity due to tsunami"
        },
        {
            "demand_type" : "Mmax",
            "demand_unit" : [
                "m^3/s^2",
                "ft^3/s^2"
            ],
            "description" : ""
        }
    ],
    "flood" : [
        {
            "demand_type" : "inundationDepth",
            "demand_unit" : [
                "ft",
                "m"
            ],
            "description" : "Depth of the water surface relative to local ground level"
        },
        {
            "demand_type" : "waterSurfaceElevation",
            "demand_unit" : [
                "ft",
                "m"
            ],
            "description" : "Elevation of the water surface above reference datum (e.g. NAVD88, mean sea level)"
        }
    ],
    "tornado" : [
        {
            "demand_type" : "wind",
            "demand_unit" : [
                "mps",
                "mph"
            ],
            "description" : "Defined as a wind velocity below"
        }
    ],
    "hurricaneWindfield" : [
        {
            "demand_type" : "3s",
            "demand_unit" : [
                "kph",
                "mph",
                "kt"
            ],
            "description" : "Typically, reported at 10 m above local ground or sea level"
        },
        {
            "demand_type" : "60s",
            "demand_unit" : [
                "kph",
                "mph",
                "kt"
            ],
            "description" : "Typically, reported at 10 m above local ground or sea level"
        }
    ],
    "hurricane" : [
        {
            "demand_type" : "waveHeight",
            "demand_unit" : [
                "ft",
                "m"
            ],
            "description" : " Height of wave measured crest to trough.  Characteristic wave height is typically the  average of one third highest waves for a random sea."
        },
        {
            "demand_type" : "surgeLevel",
            "demand_unit" : [
                "ft",
                "m"
            ],
            "description" : "Elevation of the water surface above reference datum (e.g. NAVD88, mean sea level)"
        },
        {
            "demand_type" : "inundationDuration",
            "demand_unit" : [
                "hr"
            ],
            "description" : "Time that inundation depth exceeds a critical threshold for a given storm"
        },
        {
            "demand_type" : "inundationDepth",
            "demand_unit" : [
                "ft",
                "m"
            ],
            "description" : "Depth of the water surface relative to local ground level"
        },
        {
            "demand_type" : "wavePeriod",
            "demand_unit" : "sec",
            "description" : "Time between wave crests.  Characteristic wave period is typically the inverse of the spectral peak frequency for a random sea"
        },
        {
            "demand_type" : "waveDirection",
            "demand_unit" : [
                "deg"
            ],
            "description" : "Principle wave direction associated with the characteristic wave height and period"
        },
        {
            "demand_type" : "waterVelocity",
            "demand_unit" : [
                "ft/sec",
                "m/s"
            ],
            "description" : ""
        },
        {
            "demand_type" : "windVelocity",
            "demand_unit" : [
                "mph",
                "m/sec"
            ],
            "description" : ""
        }
    ]
}
```
